### DISCUSSION TITLE
[knitr] Editors for knitr: Work with Texmaker and WinEdt, etc | /knitr/demo/editors/

### DISCUSSION THREAD

**Discussion by yihui** [rbind/yihui#753] [Comments]
https://yihui.org/knitr/demo/editors/

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
I mentioned two editors here. Which one are you talking about? And what command did you put in?

> Originally posted on 2012-04-12 17:56:52

**Reply by yihui** [reply]
You need to quote the path to Rscript because it contains a space (please take a closer look at what I wrote there). Spaces are often disasters to commands, and if you know well about the PATH variable, I'd recommend you to use PATH so that you do not have to write the full path.

> Originally posted on 2012-04-12 20:43:40

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
You mean the same Rnw file succeeded in RStudio but not in TeXmaker? Can you try a minimal example like this first? <https://github.com/yihui/knitr-examples/blob/master/002-minimal.Rnw>

> Originally posted on 2012-09-21 03:53:11

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
My suggestion is use a decent editor which takes care of all these details (e.g. RStudio). You can certainly modify that shell script, but I feel it is overly complicated.

> Originally posted on 2012-09-24 19:03:14

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Great. Thanks for sharing it.

> Originally posted on 2013-02-05 23:05:14

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks! You are right if you want to use any objects/functions in knitr (e.g. opts_chunk$set) inside the Rnw document, otherwise I do not think library(knitr) is necessary.

> Originally posted on 2013-03-30 03:00:28

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
What you mentioned is pretty much what you are supposed to do: <http://yihui.name/knitr/demo/child/>

I do not use TeXStudio, and I have no idea about the TeXStudio support of child documents. Sorry.

> Originally posted on 2013-04-04 23:14:29

**Reply by yihui** [reply]
LyX: <http://yihui.name/en/2012/10/lyx-vs-latex/>

> Originally posted on 2013-04-05 02:07:07

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Thanks! You can also call library(knitr) inside your Rnw document.

> Originally posted on 2013-05-31 20:10:36

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
That indeed looks cleaner. I'll add your method to this page. Thanks a lot!

> Originally posted on 2013-07-27 04:07:18

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Awesome. Thanks for the tip!

> Originally posted on 2014-07-22 04:08:06

[a comment by an automated account is not shown]

**Comment by HankNu** [comment]
What is wrong here?

```latex
\documentclass{article}
%\usepackage[T1]{fontenc}
\usepackage[fontset=windows]{ctex}
\begin{document}
	
Here is a code chunk.

<<foo, fig.height=4>>=
1+1
letters
chartr('xie', 'XIE', c('xie yihui', 'Yihui Xie'))
par(mar=c(4, 4, .2, .2)); plot(rnorm(100))
@

You can also write inline expressions, e.g. $\pi=\Sexpr{pi}$, and \Sexpr{1.598673e8} is a big number.

When using ctex package, this demo can be compiled in RStudio by knitr,
but failed to be compiled in TeXstudio by knitr。

e.g. 中文

\end{document}
```

**Reply by HankNu** [reply]
I find the solution:
When compiled in TeXstudio, you should configure

```sh
Rscript.exe -e "knitr::knit2pdf('%.Rnw', compiler = 'xelatex')" 
```

Please notice: knit2pdf using tinytex instead of TeXLive or something else.
